src/main/java/org/truffleruby/core/queue/SizedQueue.java

```diff
@@ -127,6 +127,28 @@ public boolean put(Object item) throws InterruptedException {
         }
     }
 
+    @TruffleBoundary
+    public Object put(Object item, long timeoutMilliseconds) throws InterruptedException {
+        lock.lock();
+        try {
+            if (size == capacity) {
```

Comment:
There should always be a loop around await(), because of spurious wakeups, i.e., await() might return but the condition is not necessarily true after it returns.

Reply:
True. Added it. (Including SizedQueue#pop and UnsizedQueue#pop too.)